src/core/store.ts

```diff
@@ -103,6 +106,16 @@ listenerMiddleware.startListening({
   },
 });
 
+listenerMiddleware.startListening({
+  actionCreator: eventCreated,
+  effect: (action) => {
+    const event = action.payload;
+    Router.push(
+      `/organize/${event.organization?.id}/projects/${event.campaign?.id}/events/${event.id}`
```

Comment:
This breaks when creating an event that isn't part of a project, like this:

1. Go to calendar
2. Go to week view
3. Drag to create a block in the calendar
4. Click "Create single event"

Expected results:
Should navigate to new event page.

Actual results:
Navigates to new event page, but then also quickly after navigates to URL where campaign ID is `undefined` (e.g. /organize/1/projects/undefined/events/82).

There is already code on `main` to redirect to a newly created event after creating it in the calendar. That code does not use this middleware approach. I'm not sure which I prefer, but regardless, there should only be one piece of code that does the same thing (right now two redirects are happening).

And regardless of where the redirect happens, it needs to account for events outside of projects. The easiest way to do that is to use the `getEventUrl()` utility function.

Reply:
Aha that was why there was no middleware for eventCreated! Thank you!